File: java/app/src/main/java/net/fbvictorhugo/j/barreirasanitaria/data/dao/IBarreiraSanitariaDAO.java
package net.fbvictorhugo.j.barreirasanitaria.data.dao;

import net.fbvictorhugo.j.barreirasanitaria.data.model.BarreiraSanitaria;

import java.util.List;

public interface IBarreiraSanitariaDAO {

    List<BarreiraSanitaria> listar();

    void inserir(BarreiraSanitaria barreiraSanitaria) throws Exception;

    void atualizar(BarreiraSanitaria barreiraSanitaria) throws Exception;

}

File: kotlin/app/src/main/java/net/fbvictorhugo/k/barreirasanitaria/data/dao/IBarreiraSanitariaDAO.kt
package net.fbvictorhugo.k.barreirasanitaria.data.dao

import net.fbvictorhugo.k.barreirasanitaria.data.model.BarreiraSanitaria

interface IBarreiraSanitariaDAO {

    fun listar(): ArrayList<BarreiraSanitaria>

    @Throws(Exception::class)
    fun inserir(barreiraSanitaria: BarreiraSanitaria)

    @Throws(Exception::class)
    fun atualizar(barreiraSanitaria: BarreiraSanitaria)

}
